vm/vm_test.go

```diff
@@ -113,6 +113,41 @@ var instructions = []struct {
 		[]interface{}{2, 2},
 		[]interface{}{},
 		thread{pc: 0, match: false, matches: map[int][]string{}}},
+	{"cmp gt float float",
+		instr{cmp, 1},
+		[]*regexp.Regexp{},
+		[]string{},
+		[]interface{}{"2.0", "1.0"},
```

Comment:
Can you convert these tests from strings to floats, and ints later on?  I think we're tripping over an accidental cast from string here.  The previous test case uses ints.